torch/_higher_order_ops/cond.py

```diff
@@ -441,16 +420,40 @@ def cond_fake_tensor_mode(mode, pred, true_fn, false_fn, operands):
     if len(flat_true_outs) != len(flat_false_outs):
         raise RuntimeError("Unmatched number of outputs from cond() branches.")
 
+    merged_outs = []
     for true_out, false_out in zip(flat_true_outs, flat_false_outs):
-        true_meta = _extract_tensor_metadata(true_out)
-        false_meta = _extract_tensor_metadata(false_out)
-        if true_meta != false_meta:
-            raise torch._dynamo.exc.CondOpArgsMismatchError(
-                f"Expected each tensor to have same metadata but got:"
-                f"\n  {true_fn.__name__} returns {true_meta}"
-                f"\n  {false_fn.__name__} returns {false_meta}"
-            )
-    return true_outs
+        merged_outs.append(_merge_tensors(true_out, false_out, mode))
+    return merged_outs
+
+
+def _merge_tensors(a: torch.Tensor, b: torch.Tensor, mode: FakeTensorMode):
+    torch._check(a.dtype == b.dtype)
+    torch._check(a.device == b.device)
+    torch._check(a.dim() == b.dim())
+
+    merged_size = []
+    for s0, s1 in zip(a.size(), b.size()):
+        # TODO: use the symint eq by expr class after rebase
+        if type(s0) is int and type(s1) is int and s0 == s1:
+            merged_size.append(s0)
+        elif (
+            isinstance(s0, torch.SymInt)
+            and isinstance(s1, torch.SymInt)
+            and s0.node.expr == s1.node.expr
+        ):
+            merged_size.append(s0)
+        else:
+            merged_size.append(mode.shape_env.create_unbacked_symint())
+            # TODO: setup value range
```

Comment:
In particular, setup a value range of min(args), max(args) (works if the inputs are symbolic or not!)